src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -1008,6 +1022,24 @@ private void FlushOpenTelemetry(IHost host)
             }
         }
 
+        private ScriptHostStartupOperation BeginStartupOperation(CancellationToken parentToken, Guid? parentId = null)
+        {
+            var operation = new ScriptHostStartupOperation(parentToken, parentId);
```

Comment:
For clarity, can you plase document the scenarios where `parentId` would be `null`?

Reply:
We only use a parent id when we're re-trying due to some other startup failure (like [here](https://github.com/Azure/azure-functions-host/blob/46194b9893def8e31a76a69e6dfbfeac5530834f/src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs#L499)). It lets us flow back via logs to where everything started and follow the chain of startup attempts. Most of the time this will be `null` as it was not created as a result of some other failure.

None of this has changed since it was introduced 6 years ago, btw

Reply:
Added some docs in a few places to note that these come in via errors only.